### DISCUSSION TITLE
Jpg instead of png

### DISCUSSION THREAD

**Discussion by chuckkay** [AIrjen/OneButtonPrompt#62] [Q&A]
Im using Vlad Diffusion with default file type output of jpg, which stores prompt in the file detail better then being hidden in png files types.. anyway to change the output default from png to jpg with one button click.
one button click should default to the systems image file type

**Comment by AIrjen** [comment]
Hi! I've been looking at the code. While I do think it is possible with some work to store the file as a .jpg, (just convert it on save) I don't think it is possible to store the prompt and generation info in it from the API.
The API supports a functin called "png-info" which is now used to gather the generation info. In thruth, this already works worse than it does when using the internal processing of either WebUI/Vlad.
I can see in the internal processing, for JPG files, it has a "dump" of the generation info.

When using the API's to generate images, this information does not seem available.

So while it may be possible to use the systems image file type, it will probably lose the ability to save the generation info. This will then also break One Button Run and upscale, since it needs that info stored in there.

**Comment by AIrjen** [comment]
I was thinking, I do write some of the generation info back in a text file during processing. And I think we do have some control over what we write back into the .PNG.
Maybe instead of going .JPG we might be able to improve on whats written into the .PNG? (While not breaking anything :D )

I've done some testing just now (haven't even used the jpg's until now)
Personally, I like the PNG files better because I can just use Notepad++ to get the generation info out. With the .jpg's I either need to drag the file to the prompt field, or have an external tool installed to read the info from the .jpg file.

Anyway, just wanted to open this discussion.

**Reply by chuckkay** [reply]
Thanks for your research, FYI jpg prompt info is stored in the exif comments section of jpg files which is easily accessed in windows properties.

![Screenshot 2023-06-30 122322](https://github.com/AIrjen/OneButtonPrompt/assets/10617746/98cb977e-45ab-4fe5-b677-2130d6d101fb)

**Comment by chuckkay** [comment]
oh one more thing that should be the convincer for you...
you can see the prompt in windows explorer by hovering over icon if you make this change in regedit
change 'Keywords' to 'Comment' in the infoTip section of [HKEY_CLASSES_ROOT\SystemFileAssociations\.jpg]
like this 
"InfoTip"="prop:System.ItemType;System.Photo.DateTaken;System.Comment;*System.Photo.PeopleNames;System.Rating;*System.Image.Dimensions;*System.Size;System.Title"
and then hover over jpg files in windows explorer and get this

![Screenshot 2023-06-30 125008](https://github.com/AIrjen/OneButtonPrompt/assets/10617746/209521ed-a314-4260-8769-8f89a50b7c7a)

**Reply by AIrjen** [reply]
Ohhh, now that is sexy! I will put this on the todo list and do research into this. It's not highest on the list currently, but this does make a good case to consider doing this. :)

**Reply by chuckkay** [reply]
FYI i informed Vlad of the regedit trick, and he is also excited and is making changes to vlad diffusion so that you dont need to change anything in regedit, and he will be adding mor features like choosing what else goes in EXIF metadata of jpgs such as prompts, sampler, etc.

https://github.com/vladmandic/automatic/discussions/1535#discussioncomment-6328856
